app/policies/department_policy.rb

```diff
@@ -0,0 +1,50 @@
+class DepartmentPolicy < ApplicationPolicy
+  class Scope
+    attr_reader :user, :department
+    def initialize(user, department)
+      @user = user
+      @department = department
+    end
+
+    def resolve
+      if @user.superadmin?
+        @department.all
+      else
+        @department.where(user: @user)
+      end
+    end
+  end
+
+  def initialize(user, department)
+    @user = user
+    @department = department
+  end
+
+  def index?
+    @user.superadmin?
+  end
+
+  def show?
+    @user.superadmin_or_department_manager?
+  end
+
+  def new?
+    @user.superadmin?
+  end
+
+  def edit?
+    @user.superadmin?
```

Comment:
this will be handled with the follow up story to this. I definied it like this, because the story said that only superadmin should be able to show the link